mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -5238,6 +5238,13 @@ static LogicalResult verifyVectorShapeCast(Operation *op,
     if (!isValidShapeCast(resultShape, sourceShape))
       return op->emitOpError("invalid shape cast");
   }
+
+  // Check that (non-)scalability is preserved
+  if (sourceVectorType.getNumScalableDims() !=
+      resultVectorType.getNumScalableDims())
+    return op->emitOpError("non-matching scalable dims");
```

Comment:
You could make it more descriptive and say what is non-matching, e.g.:
`non-matching number of scalable dims at source (2) and result (1)`